### ISSUE TITLE
JedisConnectionException when running several test cases 

### ISSUE THREAD

**Issue by Elisabeth-Engel** [kstyrc/embedded-redis#12]
Hi, 

thanks for your libary! It works great! 

I only get an error, if I run several tests directly after another. The exception looks like this:  

redis.clients.jedis.exceptions.JedisConnectionException: It seems like server has closed the connection.
	at redis.clients.util.RedisInputStream.readLine(RedisInputStream.java:90)
	at redis.clients.jedis.Protocol.processBulkReply(Protocol.java:154)
	at redis.clients.jedis.Protocol.process(Protocol.java:137)
	at redis.clients.jedis.Protocol.read(Protocol.java:200)
	at redis.clients.jedis.Connection.readProtocolWithCheckingBroken(Connection.java:285)
	at redis.clients.jedis.Connection.getStatusCodeReply(Connection.java:184)
	at redis.clients.jedis.BinaryJedis.set(BinaryJedis.java:82)
	at org.springframework.data.redis.connection.jedis.JedisConnection.set(JedisConnection.java:1092)

For one class I can help myself with switching to @BeforeClass. But then I run into the same problem, when running several test classes directly after one another. 

Do you know what might help?

**Comment by revviktor** [comment]
Hello friends,

I faced and solved the same issue.
The problem is within spring's connection.
It stores the connection, because has the same instance injected into several tests, but the server is restarted, so the connection is bad.
I found the following solution:


	@Before
	public void setUp() {
		//Needed in case of server restart
		redisDB.redis.getConnectionFactory().getConnection();
	}

redis is an instance of org.springframework.data.redis.core.StringRedisTemplate(this line is not java code :) )
